### DISCUSSION TITLE
How to create a simple Scratch AddOns (step-by-step tutorial)

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#1158] [Help] [answered]
Is there any step-by-step tutorial around on how to create a simple/dummy Scratch AddOns?

**Comment by mxmou** [comment, marked as answer]
Yes, [on the wiki](https://github.com/ScratchAddons/ScratchAddons/wiki/Creating-an-addon).

**Reply by WorldLanguages** [reply]
Keep in mind it won't be easy to start without some basic knowledge of HTML, JavaScript and CSS, and some experience with userscripts, userstyles or writing extensions.

**Reply by sfederici** [reply]
Yes, you are right. I exaggerated saying "dummy" :-) I would like to see something very simple acting on the real Scratch GUI elements (I'm mostly interested in the editor)

**Reply by sfederici** [reply]
> it won't be easy to start without some basic knowledge of HTML, JavaScript and CSS, and some experience with userscripts, userstyles or writing extensions.

I worked on the Snap's code (a Scratch clone, by Berkeley) and in the past I worked on PHP websites

**Reply by sfederici** [reply]
I got my very first look at Scratch3's code by looking at the "stage left" addon. And I realized that Scratch3's code is pretty ugly! Is it (partly) generated automatically? Or are "codes" such as "6ltCS" somewhat meaningful? I'm used to Snap's code that is perfectly logical and self-explanatory :-)

**Reply by mxmou** [reply]
> I got my very first look at Scratch3's code by looking at the "stage left" addon. And I realized that Scratch3's code is pretty ugly! Is it (partly) generated automatically? Or are "codes" such as "6ltCS" somewhat meaningful? I'm used to Snap's code that is perfectly logical and self-explanatory :-)

Yes, the stylesheets are automatically generated from SCSS files - suffixes such as _6ltCS are added to avoid conflicts. This isn't important if you're modifying the source code directly, but extensions have to deal with that.
